packages/components/src/templates/next/components/internal/Filter/FilterDrawer.tsx

```diff
@@ -0,0 +1,176 @@
+      <form className="flex-1 px-6 md:px-10">
+        {filters.map(({ id, label, items }) => (
+          <CheckboxGroup
+            className="border-b border-b-divider-medium py-4 last:border-0"
+            key={id}
+            value={holdingFiltersById[id] ?? []}
+            onChange={(values) => {
+              setHoldingFiltersById((prev) => ({
+                ...prev,
+                [id]: values,
+              }))
+            }}
+          >
+            <AriaButton
+              className={composeRenderProps("", (className, renderProps) =>
+                expandFilterButtonStyle({
+                  ...renderProps,
+                  className,
+                }),
+              )}
+              onPress={() => updateFilterToggle(id)}
+            >
+              <Label>{label}</Label>
+              <BiChevronDown
+                aria-hidden
+                className={`mr-3 h-6 w-6 text-base-content-strong transition-all duration-300 ease-in-out ${
+                  showFilter[id] ? "rotate-180" : "rotate-0"
+                }`}
+              />
+            </AriaButton>
+
+            <div className={showFilter[id] ? "flex flex-col" : "hidden"}>
+              {items.map(({ id: itemId, label: itemLabel, count }) => (
+                <Checkbox
+                  value={itemId}
+                  key={itemId}
+                  className="w-fit cursor-pointer p-2"
+                >
+                  {itemLabel} ({count.toLocaleString()})
+                </Checkbox>
+              ))}
+            </div>
+          </CheckboxGroup>
+        ))}
+      </form>
+      {/* Sticky action bottom bar */}
+      <div className="sticky bottom-0 left-0 right-0 flex flex-col gap-3 border-t border-t-divider-medium bg-white px-6 pb-12 pt-8 md:px-10">
+        <Button
+          className="w-full"
+          variant="solid"
+          size="lg"
+          onPress={handleApplyFilters}
+        >
+          Apply filters
+        </Button>
+        <Button
+          size="lg"
+          className="w-full"
+          variant="outline"
+          onPress={handleClearFilter}
+        >
+          Clear all filters
+        </Button>
+      </div>
+    </>
+  )
+}
+
+export const FilterDrawer = (props: FilterDrawerProps): JSX.Element => {
+  const { isOpen, onOpen } = props
+
+  return (
+    <Dialog open={isOpen} onClose={onOpen} className="relative z-40 lg:hidden">
+      <DialogBackdrop
+        transition
+        className="fixed inset-0 bg-black bg-opacity-25 transition-opacity duration-300 ease-linear data-[closed]:opacity-0"
+      />
+
+      <div className="fixed inset-0 z-40 flex">
+        <DialogPanel
+          transition
+          className="relative ml-auto flex h-full w-full transform flex-col overflow-y-auto bg-white transition duration-300 ease-in-out data-[closed]:translate-y-full"
+        >
+          <div className="mx-6 flex items-center justify-between border-b border-b-divider-medium pb-3 pt-12 md:mx-10">
```

Comment:
nit: feels a little spacious at the top

<img width="379" alt="Screenshot 2024-08-16 at 11 16 40" src="https://github.com/user-attachments/assets/c66eca9b-d3a0-4fed-884e-56eaf13fa09e">

Reply:
this one is design liow

Reply:
FYI yeah @sehyunidaaa but not blocking this PR